src/game/requester.c

```diff
@@ -174,34 +185,61 @@ void Requester_SetHeading(REQUEST_INFO *req, const char *string)
     Text_Remove(req->heading);
     req->heading = NULL;
 
-    if (string) {
-        req->heading_text = string; // unsafe
-    } else {
-        req->heading_text = NULL;
-    }
+    const size_t out_size = snprintf(NULL, 0, "%s", string) + 1;
+    snprintf(req->heading_text, out_size, "%s", string);
```

Comment:
Should I also use malloc here in `Requester_SetHeading`? It's the only init malloc left using ` eq->item_text_len`.

Reply:
I would say so, yes.